packages/next/client/script.tsx

```diff
@@ -78,14 +79,29 @@ const loadScript = (props: ScriptProps): void => {
   LoadCache.add(cacheKey)
 
   if (dangerouslySetInnerHTML) {
-    el.innerHTML = dangerouslySetInnerHTML.__html || ''
+    console.error(
+      "Using 'dangerouslySetInnerHTML' within <script> tags is not supported by Next.js."
+    )
   } else if (children) {
-    el.textContent =
-      typeof children === 'string'
-        ? children
-        : Array.isArray(children)
-        ? children.join('')
-        : ''
+    // Assign to el.textContent based on the type of children
+    if (typeof children === 'string' || isTrustedScript(children)) {
+      el.textContent = children as string
+    } else if (Array.isArray(children)) {
+      let textContent: TrustedScript | string = children.join('')
+
+      // If all of the children are TrustedScripts or if textContent is empty,
+      // then it is safe to promote textContent to a TrustedScript
+      if (
+        children.every((child) => isTrustedScript(child)) ||
+        textContent === ''
```

Comment:
What's the motivation for the `textContent === ''` check? Note that if the children array is empty, then the `children.every` condition will be true.

Reply:
This check has now been removed.